### PR TITLE
160525336 Get post occupations

### PR DESCRIPTION
**Pull request by dmaish**
#### What does this PR do?
Fetches and adds new occupations to the DB

#### Description of Task to be completed?
When the user is making a new request, To ensure ease of use, it is important for users to be able to search and select their current roles from a pre-populated list that is dynamically updated as the user types in their role.

#### How should this be manually tested?
branch: ft-get-post-occupations-160525336

- Clone the repo - ```git clone https://github.com/andela/travel_tool_front.git```
- Change into the project directory and run git checkout ```ft-get-post-occupations-160525336```
- cd into the branch ```ft-get-post-occupations-160525336```
- run ```yarn run db:rollmigrate``` to add the existing occupation from the seeders to the db
- Run the server using ```yarn run start:dev``` or ```make start```
- use postman to send requests using the above endpoints
      ```POST /api/v1/occupations``` to add occupations to the db
      ```GET /api/v1/occupations``` to list all occupations

#### Any background context you want to provide?
- at the moment, the requirement for the addition of the occupations through the POST endpoint haven't been decided on yet.This means any user can add an occupation to the db.It wasn't also a scope of this task though it was essential in getting it to work.

#### What are the relevant pivotal tracker stories?
[160525336](https://www.pivotaltracker.com/story/show/160525336)

#### Screenshots (if appropriate)
screen shot 2018-10-04 at 9 58 13 am

### CODE REVIEW

File: src/database/migrations/20180921005932-create-occupation.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.createTable('Occupations', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      occupationName: {
+        type: Sequelize.STRING
```
Comment (Collins33) [inline src/database/migrations/20180921005932-create-occupation.js]: On the occupation name, the user can put a null name? You should change that

File: src/modules/occupations/OccupationsController.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  static async createOccupations(req, res) {
+    try {
+      const { occupationName } = req.body;
+      if (!occupationName) {
+        return res.status(400).json({
+          request: JSON.stringify(req.body),
+          success: false,
+          message: ' new occupationName name required '
```
Comment (Collins33) [inline src/modules/occupations/OccupationsController.js]: You could change this response to New occupation name required

File: src/modules/occupations/__tests__/occupations.test.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  it('should add a new occupation to the db', async (done) => {
+    request(app)
+      .post('/api/v1/occupations')
+      .set('Content-Type', 'application/json')
+      .set('authorization', token)
+      .send(newOccupation)
+      .expect(201)
+      .end((err, res) => {
+        expect(res.status).toEqual(201);
+        // expect(res.body).toEqual(true);
```
Comment (Collins33) [inline src/modules/occupations/__tests__/occupations.test.js]: Remove commented code.

File: src/modules/occupations/__tests__/occupations.test.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+  it('should add a new occupation to the db', async (done) => {
+    request(app)
+      .post('/api/v1/occupations')
+      .set('Content-Type', 'application/json')
+      .set('authorization', token)
+      .send(newOccupation)
+      .expect(201)
+      .end((err, res) => {
+        expect(res.status).toEqual(201);
```
Comment (daddychukz) [inline src/modules/occupations/__tests__/occupations.test.js]: Don't just test for status codes, also test for the values sent if it's expected. Do this for other tests you have as well

Comment (dmaish) [reply src/modules/occupations/__tests__/occupations.test.js]: already fixed this.Messages from the controller are being asserted too -success statuses messages to the user.

File: src/modules/occupations/__tests__/occupations.test.js (hunk 3 above)
Comment (dmaish) [reply src/modules/occupations/__tests__/occupations.test.js]: fixed

File: src/modules/occupations/OccupationsController.js (hunk 2 above)
Comment (dmaish) [reply src/modules/occupations/OccupationsController.js]: done this.

File: src/database/migrations/20180921005932-create-occupation.js (hunk 1 above)
Comment (dmaish) [reply src/database/migrations/20180921005932-create-occupation.js]: allow null changed to false.no null value can be saved now.

**Pull request merged**